reference/tools/data-migration/features/online-ddl-scheme.md

```diff
@@ -0,0 +1,195 @@
+    ```
+
+- 不执行 `_test4_gho` 的 DDL 操作。把执行的 DDL 记录到 dm_meta.{task_name}\_onlineddl 以及内存中。
+
+    ```sql
+    REPLACE INTO dm_meta.{task_name}_onlineddl (id, ghost_schema , ghost_table , ddls) VALUES (......);
+    ```
+
+- 只要不是 **realtable** 的 DML 全部不执行。
+- rename 拆分成两个 SQL。
+
+    ```sql
+    rename test.test4 to test._test4_del;
+    rename test._test4_gho to test.test4;
+    ```
+
+- 不执行 `rename to _test4_del`。当要执行 `rename ghost_table to origin table` 的时候，并不执行 rename，而是把步骤 3 内存中的 DDL 读取出来，然后把 ghost_table、ghost_schema 替换为 origin_table 以及对应的 schema 再执行。
+
+    ```sql
+    alter table test._test4_gho add column cl1 varchar(20) not null;
+    --替换为
+    alter table test.test4 add column cl1 varchar(20) not null;
+    ```
+
+## online-schema-change: pt
+
+pt-osc 在实现 online-schema-change 的过程会产生 2 种 table：
+
+- **new** 用于应用 DDL，待数据同步追上 origin table 之后会通过 rename 的方式替换 origin table。
+- **old** 是由 origin table rename 过来的。
+- 3 种 **trigger**：pt_osc\_\*\_ins、pt_osc\_\*\_upd、pt_osc\_\*\_del，用于在 pt_osc 过程中，同步 origin table 新产生的数据到 new。
+
+DM 在同步过程中会把上述 table 分成 3 类：
+
+- ghostTable : \_\*\_new
+- trashTable : \_\*\_old
+- realTable : 执行的 online-ddl 的 origin table
+
+pt-osc 主要涉及的 SQL如下：
+
+``` sql
+-- 1.
+   CREATE TABLE `test`.`_test4_new` ( id int(11) NOT NULL AUTO_INCREMENT,
+   date date DEFAULT NULL, account_id bigint(20) DEFAULT NULL, conversion_price decimal(20,3) DEFAULT NULL,  ocpc_matched_conversions bigint(20) DEFAULT NULL, ad_cost decimal(20,3) DEFAULT NULL,cl2 varchar(20) COLLATE utf8mb4_bin NOT NULL,cl1 varchar(20) COLLATE utf8mb4_bin NOT NULL,PRIMARY KEY (id) ) ENGINE=InnoDB AUTO_INCREMENT=3 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ;
+-- 2.
+   ALTER TABLE `test`.`_test4_new` add column c3 int
+-- 3.
+   CREATE TRIGGER `pt_osc_test_test4_del` AFTER DELETE ON `test`.`test4` ...... ;
+   CREATE TRIGGER `pt_osc_test_test4_upd` AFTER UPDATE ON `test`.`test4` ...... ;
+   CREATE TRIGGER `pt_osc_test_test4_ins` AFTER INSERT ON `test`.`test4` ...... ;
+-- 4.
+   INSERT LOW_PRIORITY IGNORE INTO `test`.`_test4_new` (`id`, `date`, `account_id`, `conversion_price`, `ocpc_matched_conversions`, `ad_cost`, `cl2`, `cl1`) SELECT `id`, `date`, `account_id`, `conversion_price`, `ocpc_matched_conversions`, `ad_cost`, `cl2`, `cl1` FROM `test`.`test4` LOCK IN SHARE MODE /*pt-online-schema-change 3227 copy table*/
+-- 5.
+   RENAME TABLE `test`.`test4` TO `test`.`_test4_old`, `test`.`_test4_new` TO `test`.`test4`
+-- 6.
+   DROP TABLE IF EXISTS `test`.`_test4_old`;
+   DROP TRIGGER IF EXISTS `pt_osc_test_test4_del` AFTER DELETE ON `test`.`test4` ...... ;
+   DROP TRIGGER IF EXISTS `pt_osc_test_test4_upd` AFTER UPDATE ON `test`.`test4` ...... ;
+   DROP TRIGGER IF EXISTS `pt_osc_test_test4_ins` AFTER INSERT ON `test`.`test4` ...... ;
+```
+
+> **注意：**
+>
+> 具体 pt-osc 的 SQL 会根据工具执行时所带的参数而变化。本文只列出主要的 SQL ，具体可以参考 pt-osc 官方文档。
+
+### DM 对于 **online-ddl-scheme: pt** 的处理
```

Comment:
@SE-Bin The same question applies to this section. But there are 7 DM operations while only 6 SQL statements above. Can we also put each operation under each SQL? If so, how to address the additional one?